### ISSUE TITLE
TC39 proposal - Error.cause property

### ISSUE THREAD

**Issue by sirian** [microsoft/TypeScript#45167]
# lib.es2021d.ts Update Request - Error.cause property

Proposal (Stage 3)
https://github.com/tc39/proposal-error-cause

Already implemented in Chrome 93: 
https://www.chromestatus.com/feature/5727099325251584

## Sample Code
```typescript
interface Error {
    cause?: any;
}

interface ErrorInit {
    cause?: any;
}

interface ErrorConstructor {
    new(message?: string, init?: ErrorInit): Error;
    (message?: string, init?: ErrorInit): Error;
}
```

**Comment by johanneswuerbach** [comment]
`Error.cause` is also available in [Node 16.9.0](https://github.com/nodejs/node/releases/tag/v16.9.0) now 🎉

**Comment by Alexsey** [comment]
Already Stage 4 - the feature is finalized!

**Comment by sandersn** [comment]
Fixed by #46291 (and #45749, but I missed reviewing that one.)

**Comment by amakhrov** [comment]
This seems to be partially implemented.

The constructor now accepts `ErrorInit` with the `cause` property - which is great.
However, this property is not declared in the Error object itself.

With the full implementation this example should compile:
```
const error = new Error("derived", {cause: new Error("original")}); // OK
console.log(error.cause); // Property 'cause' doesn't exist
```
@sandersn

**Comment by sandersn** [comment]
Thanks for pointing that out. I re-opened the bug.

**Comment by GoudekettingRM** [comment]
Bump. Also ran into this. Fixed it local by redeclaring like this:
```ts
interface Error {
  name: string;
  message: string;
  cause?: any;
  stack?: string;
}

interface ErrorConstructor {
  new(message?: string, { cause }?: { cause: any }): Error;
  (message?: string): Error;
  readonly prototype: Error;
}

declare var Error: ErrorConstructor
```

**Comment by mcaskill** [comment]
I also had this issue for `cause` and `stack`. Fixed it by redeclaring like this:

```ts
interface ErrorInit {
    cause?: unknown;
}

declare type BaseError = Error;

declare class Error implements BaseError {
    name: string;
    message: string;
    stack?: string;
    cause?: unknown;

    constructor(message?: string, init?: ErrorInit);

    // eslint-disable-next-line @typescript-eslint/ban-types -- Allow catch-all 'function' parameter type.
    static captureStackTrace(error: object, constructor?: Function): void;
}
```

**Comment by yume-chan** [comment]
`cause` was added in #47020 and all definitions had shipped with 4.6.

**Comment by MartinJohns** [comment]
@yume-chan Although the type is inaccurate, as per #48098.
